app/database/operator/server_data_operator/handlers/channel.ts

```diff
@@ -130,23 +131,28 @@ const ChannelHandler = (superclass: any) => class extends superclass {
      * @throws DataOperatorException
      * @returns {Promise<MyChannelModel[]>}
      */
-    handleMyChannel = ({channels, myChannels, prepareRecordsOnly = true}: HandleMyChannelArgs): Promise<MyChannelModel[]> => {
+    handleMyChannel = async ({channels, myChannels, prepareRecordsOnly = true}: HandleMyChannelArgs): Promise<MyChannelModel[]> => {
         if (!myChannels.length) {
             throw new DataOperatorException(
                 'An empty "myChannels" array has been passed to the handleMyChannel method',
             );
         }
 
+        const isCRTEnabled = await getIsCRTEnabled(this.database);
+
         const channelMap = channels.reduce((result: Record<string, Channel>, channel) => {
             result[channel.id] = channel;
             return result;
         }, {});
+
         for (const my of myChannels) {
             const channel = channelMap[my.channel_id];
             if (channel) {
-                const msgCount = Math.max(0, channel.total_msg_count - my.msg_count);
+                const total = isCRTEnabled ? channel.total_msg_count_root! : channel.total_msg_count;
+                const myMsgCount = isCRTEnabled ? my.msg_count_root! : my.msg_count;
+                const msgCount = Math.max(0, total - myMsgCount);
                 my.msg_count = msgCount;
-                my.is_unread = msgCount > 0;
+                my.is_unread = my.notify_props.mark_unread === 'all' && msgCount > 0;
```

Comment:
Can you clarify the muted state? And yes I mean the notification prop that should actually check not to be undefined?

Reply:
It seems that it's not that straight forward keeping the `is_unread` in the db like that.
It depends on a number of factors. 
The `notifyProp.markUnread` can be either 'all' or 'mention' and then it depends either on all messages or only on mentions, which in their turn depend on CRT state.

I'd like to propose to create a `isUnread` method on the model, so that we calculate on read rather on write.
Also I'd like to create a new task to solve that since it affects more than CRT, irrelevant on if we make the method or not that is.

Reply:
Go for it

Reply:
But I want it on write not on read as it simplifies way too much